tutorials/preprocessing/plot_70_fnirs_processing.py

```diff
@@ -0,0 +1,199 @@
+
+
+###############################################################################
+# Remove heart rate from signal
+# -----------------------------
+#
+# The haemodynamic response has frequency content predominantly below 0.5 Hz.
+# An increase in activity around 1 Hz can be seen that is due to the heart beat
+# and is unwanted. So we use low pass filter can be used to remove this.
+
+raw_fnirs.plot_psd(average=True)
+raw_fnirs = raw_fnirs.filter(0.05, 0.7, h_trans_bandwidth=0.2,
+                             l_trans_bandwidth=0.02)
+raw_fnirs.plot_psd(average=True)
+
+
+###############################################################################
+# Extract epochs
+# --------------
+#
+# Now that the signal has been converted to relative haemoglobin concentration,
+# and the unwanted heart rate component has been removed, we can extract epochs
+# related to each of experimental conditions.
+#
+# First we extract the events of interest and visualise them to ensure they are
+# correct.
+
+events, _ = mne.events_from_annotations(raw_fnirs, event_id={'1.0': 1,
+                                                             '2.0': 2,
+                                                             '3.0': 3})
+event_dict = {'Control': 1, 'Tapping/Left': 2, 'Tapping/Right': 3}
+fig = mne.viz.plot_events(events, event_id=event_dict,
+                          sfreq=raw_fnirs.info['sfreq'])
+fig.subplots_adjust(right=0.7)  # make room for the legend
+
+
+###############################################################################
+# Next we define the range of our epochs, the rejection criteria,
+# baseline correction, and extract the epochs. We visualise the log of which
+# epochs were dropped.
+
+reject_criteria = dict(hbo=80e-6)
+tmin, tmax = -5, 15
+
+epochs = mne.Epochs(raw_fnirs, events, event_id=event_dict,
+                    tmin=tmin, tmax=tmax,
+                    reject=reject_criteria, reject_by_annotation=True,
+                    proj=True, baseline=(None, 0), preload=True,
+                    detrend=None, verbose=True)
+epochs.plot_drop_log()
+
+
+###############################################################################
+# View consistency of responses across trials
+# -------------------------------------------
+#
+# Now we can view the haemodynamic response for our tapping condition.
+# We visualise the response for both the oxy- and deoxyhaemoglobin, and
+# observe the expected peak in HbO at around 6 seconds consistently across
+# trials, and the consistent dip in HbR that is slightly delayed relative to
+# the HbO peak.
+
+epochs['Tapping'].plot_image(combine='mean', vmin=-30, vmax=30)
+
+
+###############################################################################
+# We can also view the epoched data for the control condition and observe
+# that it does not show the expected morphology.
+
+epochs['Control'].plot_image(combine='mean', vmin=-30, vmax=30)
```

Comment:
you might consider passing `ts_args=dict(ylim=(-12, 12))` here and in the previous plot, to get the ERP plots below each image plot to all have the same axis lims.  I find it makes comparison across figures easier.  Note that `(-12, 12)` is just a guess at what might look best.

Reply:
THATS HOW YOU DO IT!!! I am embarrassed to tell you how long I spent fiddling around trying to do that.